ui/src/App.tsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect } from 'react';
+
+import './App.css';
+import {
+  makeAgoricChainStorageWatcher,
+  AgoricChainStoragePathKind as Kind,
+} from '@agoric/rpc';
+import { create } from 'zustand';
+import {
+  makeAgoricWalletConnection,
+  suggestChain,
+} from '@agoric/web-components';
+import { subscribeLatest } from '@agoric/notifier';
+import { makeCopyBag } from '@agoric/store';
+import { Logos } from './components/Logos';
+import { Inventory } from './components/Inventory';
+import { Trade } from './components/Trade';
+
+const { entries, fromEntries } = Object;
+
+const RUN_ENV = import.meta.env.VITE_RUN_ENV || 'localhost';
+if (!['localhost', 'agoric_chain'].includes(RUN_ENV))
+  throw new Error("VITE_RUN_ENV can only be 'agoric_chain' or 'localhost'");
+
+type Wallet = Awaited<ReturnType<typeof makeAgoricWalletConnection>>;
+
+const ENDPOINTS = {
+  RPC: `http://${RUN_ENV}:26657`,
+  API: `http://${RUN_ENV}:1317`,
+};
+
+const watcher = makeAgoricChainStorageWatcher(ENDPOINTS.API, 'agoriclocal');
+
+interface AppState {
+  wallet?: Wallet;
+  offerUpInstance?: unknown;
+  brands?: Record<string, unknown>;
+  purses?: Array<Purse>;
+}
+
+const useAppStore = create<AppState>(() => ({}));
+
+const setup = async () => {
+  watcher.watchLatest<Array<[string, unknown]>>(
+    [Kind.Data, 'published.agoricNames.instance'],
+    instances => {
+      console.log('got instances', instances);
+      useAppStore.setState({
+        offerUpInstance: instances.find(([name]) => name === 'offerUp')!.at(1),
+      });
+    },
+  );
+
+  watcher.watchLatest<Array<[string, unknown]>>(
+    [Kind.Data, 'published.agoricNames.brand'],
+    brands => {
+      console.log('Got brands', brands);
+      useAppStore.setState({
+        brands: fromEntries(brands),
+      });
+    },
+  );
+};
+
+const connectWallet = async () => {
+  await suggestChain(`http://localhost:3004/${RUN_ENV}`);
+  const wallet = await makeAgoricWalletConnection(watcher, ENDPOINTS.RPC);
```

Comment:
Why not add this URL as an env variable? Is this the only change we making in UI-related code? 

If this is the only change we're making then we can use a conditional to decide with what param `suggestChain` is called. The conditional can be decided based on the value of some env variable.